### PR TITLE
fix: update gitignore, detect missing func runtime

### LINKED ISSUE #7471: Add check for func runtime during creation to prevent downstream problems
## Describe the bug
Function bots don't start. There's an error saying `'func' is not recognized as an internal or external command, operable program or batch file.`

## Version
Version:  2.0.0-nightly.240366.4a6ec2c
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a dotnet echo function bot (via empty bot project template)
2. Create an echo trigger 
3. Create an echo dialog that just echos the activity
4. click start

## Expected behavior
The bot should start

## Screenshots
image

### LINKED ISSUE #7473: .gitignore for newly created bots should include generated directory
## Describe the bug

It appears that, upon creating a new bot through Composer, a .gitignore file is generated in the created project. As per the ask originally outlined in https://github.com/microsoft/botframework-components/issues/953:

[.gitignore](https://github.com/microsoft/botframework-components/blob/main/generators/generator-bot-adaptive/generators/app/templates/dotnet/webapp/.gitignore) should add an ignore setting to exclude the generated folder from source control.

```
# composer files
generated/
```

The generated folder contains files that are based on the current user settings and paths and that are not necessarily the same as the ones in others developers machines or the deployed server.

The content of this folder is regenerated every time composer runs based on other files that are tracked in source control and are not needed.

The CI/CD pipeline should also regenerate these files to make sure they point to the deployed models.

## Version

v2.0.0-nightly.240624.c8283fa

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Click the 'New' button on the Composer home screen.
2. Create a new bot using one of the presented templates.
3. Navigate to the directory containing the newly created bot.
4. Open the created .gitignore file.

## Expected behavior

.gitignore should contain the following or similar:

```
# Generated folder and contents produced by Bot Framework Composer.
generated/
```

## Screenshots

image

## Additional context

Contents of the generated .gitignore for reference:

```
# prevent appsettings.json get checked in
**/appsettings.json
```

### PR DESCRIPTION
**Pull request by benbrown**
## Description

Fixes #7473: Update the .gitignore file that is added to each project as part of the boilerplate content
Fixes #7471: Detect missing func runtime during create flow and when the bot starts. Display useful errors.

Screen Shot 2021-04-29 at 2 51 56 PM

Screen Shot 2021-04-29 at 12 20 13 PM

### CODE REVIEW

**Review by peterinnesmsft — APPROVED** [review]

**Review by hatpick — APPROVED** [review]

**Review by beyackle — DISMISSED** [review]

File: Composer/packages/client/src/utils/runtimeErrors.ts [hunk 1]
```diff
@@ -15,6 +15,23 @@ export const missingDotnetVersionError = {
...
+    url: 'https://www.npmjs.com/package/azure-functions-core-tools',
+  },
+};
+
 export const checkIfDotnetVersionMissing = (err: { message: string }) => {
   return /(Command failed: dotnet user-secrets)|(install[\w\r\s\S\t\n]*\.NET Core SDK)/.test(err.message as string);
 };
+
+export const checkIfFunctionsMissing = (err: { message: string }) => {
+  return /(Azure Functions runtime not installed.)|(spawn func ENOENT)/.test(err.message as string);
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/utils/runtimeErrors.ts:36]: `ENOENT` seems a little broad. Are we sure that all `ENOENT` errors are due to functions missing?

Comment (benbrown) [reply Composer/packages/client/src/utils/runtimeErrors.ts:36]: I am not sure -- this is the error that comes back when the `func` command is missing or not functional.

Comment (cwhitten) [reply Composer/packages/client/src/utils/runtimeErrors.ts:36]: I think we may be OK here as it explicitly checks for `func`, which is an Azure Functions thing: https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=macos%2Ccsharp%2Cbash#create-a-local-functions-project

Comment (a-b-r-o-w-n) [reply Composer/packages/client/src/utils/runtimeErrors.ts:36]: That's a good point. I just read right past that!

**Review by a-b-r-o-w-n — APPROVED** [review]

**Pull request merged**